client/src/modules/stock/exit/exit.js

```diff
@@ -47,169 +38,173 @@ function StockExitController(
+        field: 'available_lot',
+        width: 150,
+        displayName: 'TABLE.COLUMNS.AVAILABLE',
+        headerCellFilter: 'translate',
+        cellTemplate: 'modules/stock/exit/templates/available.tmpl.html',
       }, {
-        field : 'amount',
-        width : 150,
-        displayName : 'TABLE.COLUMNS.AMOUNT',
-        headerCellFilter : 'translate',
-        cellTemplate : 'modules/stock/exit/templates/amount.tmpl.html',
+        field: 'amount',
+        width: 150,
+        displayName: 'TABLE.COLUMNS.AMOUNT',
+        headerCellFilter: 'translate',
+        cellTemplate: 'modules/stock/exit/templates/amount.tmpl.html',
       }, {
-        field : 'expiration_date',
-        width : 150,
-        displayName : 'TABLE.COLUMNS.EXPIRE_IN',
-        headerCellFilter : 'translate',
-        cellTemplate : 'modules/stock/exit/templates/expiration.tmpl.html',
+        field: 'expiration_date',
+        width: 150,
+        displayName: 'TABLE.COLUMNS.EXPIRE_IN',
+        headerCellFilter: 'translate',
+        cellTemplate: 'modules/stock/exit/templates/expiration.tmpl.html',
       },
-      { field : 'actions', width : 25, cellTemplate : 'modules/stock/exit/templates/actions.tmpl.html' },
+      { field: 'actions', width: 25, cellTemplate: 'modules/stock/exit/templates/actions.tmpl.html' },
     ],
-    data : vm.Stock.store.data,
-    fastWatch : true,
-    flatEntityAccess : true,
+    data: vm.stockForm.store.data,
+    fastWatch: true,
+    flatEntityAccess: true,
   };
 
+  // exposing the grid options to the view
   vm.gridOptions = gridOptions;
-  vm.checkValidity = checkValidity;
 
   function selectExitType(exitType) {
     vm.movement.exit_type = exitType.label;
     mapExit[exitType.label].find();
-    // FIXME: textarea default value must be translated in the view
     vm.movement.description = $translate.instant(mapExit[exitType.label].description);
   }
 
   function setupStock() {
-    vm.Stock.setup();
-    vm.Stock.store.clear();
+    vm.stockForm.setup();
+    vm.stockForm.store.clear();
   }
 
   // add items
   function addItems(n) {
-    vm.Stock.addItems(n);
+    vm.stockForm.addItems(n);
     checkValidity();
   }
 
   // remove item
   function removeItem(item) {
-    vm.Stock.removeItem(item.index);
+    vm.stockForm.removeItem(item.id);
     checkValidity();
   }
 
   // configure item
   function configureItem(item) {
     item._initialised = true;
     // get lots
-    Stock.lots.read(null, { depot_uuid : vm.depot.uuid, inventory_uuid : item.inventory.inventory_uuid, includeEmptyLot : 0 })
+    Stock.lots.read(null, { depot_uuid: vm.depot.uuid, inventory_uuid: item.inventory.inventory_uuid, includeEmptyLot: 0 })
       .then(function (lots) {
         item.lots = lots;
       })
       .catch(Notify.handleError);
   }
 
   function startup() {
+    // setting params for grid loading state
+    vm.loading = true;
+    vm.hasError = false;
+    vm.message = 'STOCK.EMPTY';
```

Comment:
It doesn't look like this ever changes.  Could you define this in the HTML?  Like this:
```html
<bh-grid-loading-indicator
  ...
  message-state="'STOCK.EMPTY'">
</bh-grid-loading-indicator>
```